tests/integration/frameworks/test_transformers_unit.py

```diff
@@ -291,3 +291,83 @@ def test_custom_pipeline(pair_classification_pipeline: PairClassificationPipelin
     ) == pair_classification_pipeline("I hate you", second_text="I love you")
 
     runner.destroy()
+
+
+def test_import_model_with_synced_version():
+    revision = "3956d303d3cddf0708ff20660c1ea5f6ec30e434"
+    bento_model = bentoml.transformers.import_model(
+        "tiny-bert",
+        "hf-internal-testing/tiny-random-BertModel",
+        sync_with_hub_version=True,
+        revision=revision,
+    )
+
+    assert bento_model.tag.version == revision
+    bentoml.models.delete("tiny-bert:3956d303d3cddf0708ff20660c1ea5f6ec30e434")
+
+    revision = "3956d303d3cddf0708ff20660c1ea5f6ec30e434"
+    bento_model = bentoml.transformers.import_model(
+        "tiny-bert:asdf",
+        "hf-internal-testing/tiny-random-BertModel",
+        sync_with_hub_version=True,
+        revision=revision,
+    )
+
+    assert bento_model.tag.version == revision
+    bentoml.models.delete("tiny-bert:3956d303d3cddf0708ff20660c1ea5f6ec30e434")
+
+    revision = "3956d303d3cddf0708ff20660c1ea5f6ec30e434"
+    bento_model = bentoml.transformers.import_model(
+        "tiny-bert:asdf",
+        "hf-internal-testing/tiny-random-BertModel",
+        revision=revision,
+    )
+
+    assert bento_model.tag.version == "asdf"
+    bentoml.models.delete("tiny-bert:asdf")
+
+
+def test_import_model_has_required_files():
+    revision = "3956d303d3cddf0708ff20660c1ea5f6ec30e434"
+    labels = {"framework": "transformers"}
+    bento_model = bentoml.transformers.import_model(
+        "tiny-bert",
+        "hf-internal-testing/tiny-random-BertModel",
+        sync_with_hub_version=True,
+        revision=revision,
+        labels=labels,
+    )
+
+    assert os.path.exists(bento_model.path_of("config.json"))
+    assert os.path.exists(bento_model.path_of("pytorch_model.bin"))
+    assert os.path.exists(bento_model.path_of("pretrained.v2.pkl"))
+    assert os.path.exists(bento_model.path_of("model.yaml"))
+    assert bento_model.info.labels == labels
+    bentoml.models.delete("tiny-bert:3956d303d3cddf0708ff20660c1ea5f6ec30e434")
+
+
+def test_import_model_with_pretrained_class():
+    revision = "3956d303d3cddf0708ff20660c1ea5f6ec30e434"
+    pretrained_class = transformers.BertForMaskedLM
```

Comment:
I assume this also work for a remote model? Or trust_remote_code=True?

Reply:
Yup using trust_remote_code=True does work. But testing for this functionality actually brought up an issue that I was not aware of before. Tested with Baichuan and chatglm, it seems that they have their own predefined architecture (not one of the pretrained ones from transformers) which doesn't seem to be compatible with `init_empty_weights` from `accelerate`. 

The only work around I found was to snapshot download the repo as that does not use `.from_pretrained`, thus not loading the model weights. This is not the ideal behavior but the transformers library seems to be lacking in terms of what we want to accomplish 😢
I have been following this thread: https://discuss.huggingface.co/t/downloading-a-model-from-the-hub-without-loading-it/48377

With the new changes I added, the below should work:
`bentoml.transformers.import_model('baichuan-test', 'baichuan-inc/Baichuan-7B', trust_remote_code=True, clone_repository=True)`

Reply:
The from_pretrained should work with `trust_remote_code` path right? it is a trust_remote_code path in the meta class AutoModel.

But do we need the class itself for loading? we can just pickled the class afterwards right?

Reply:
from_pretrained does work with the `trust_remote_code` kwarg. 

However, the problem comes when `config.architectures` specifies an architecture not supported by transformers. When this happens, the repository comes with its own `modeling_xyz.py`, which defines its own pretrained model class. `init_empty_weights` is used to instantiate the model class without actually loading the weights, but it doesn't seem to be compatible with custom pretrained classes (so those that are not supported by transformers by default).

Therefore, the problem right now is that custom pretrained model classes will actually be loaded into memory. The only way to avoid it is to set the `clone_repository` kwarg. This will call snapshot download and just downloads the entire repository instead of being selective of what files we want (which from_pretrained was primarily used for).

Reply:
> But do we need the class itself for loading? we can just pickled the class afterwards right?

An instance of the class is needed for generating default signatures.

Reply:
This is a intended behaviour of trust_remote_code afaik. You don't really have any other way other than loading the model into memory right?

You have to load the model into memory with trust_remote_code to verify so basically I think it is fine.